src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -158,8 +189,44 @@ public Person addPerson(Person person) {
     @Override
     public void setPerson(Person target, Person editedPerson) {
         requireAllNonNull(target, editedPerson);
-
         eduMate.setPerson(target, editedPerson);
+        updateMeetUpForEditPerson(target, editedPerson);
+        editParticipants(target, editedPerson);
+
+    }
+
+    public void editParticipants(Person target, Person editedPerson) {
+        eduMate.editParticipants(target, editedPerson);
+    }
+
+    /**
+     * Updates scheduled meet up list for participants with edited details.
+     * @param target Person to edit.
+     * @param editedPerson Edited person to replace person to edit.
+     */
+    public void updateMeetUpForEditPerson(Person target, Person editedPerson) {
+        Iterator<MeetUp> iterator = observableMeetUps.iterator();
+        while (iterator.hasNext()) {
+            MeetUp meetUp = iterator.next();
+            Participants participants = meetUp.getParticipants();
+            List<Person> participantsList = participants.getParticipants();
+            updatePersonList(participantsList, target, editedPerson, meetUp);
+        }
+    }
+
+    /**
+     * Updates participant list of a scheduled meet up if person to edit exists.
+     * @param participantsList List of persons participating in the meet up.
+     * @param target Person to be edited.
+     * @param editedPerson Person with edited details.
+     * @param meetUp The meet up to be updated.
+     */
+    public void updatePersonList(List<Person> participantsList, Person target,
+                                 Person editedPerson, MeetUp meetUp) {
+        if (participantsList.removeIf(person -> person.isSamePerson(target))) {
+            participantsList.add(editedPerson);
+            meetUp.setParticipants(new Participants(participantsList));
```

Comment:
Be careful of using `removeIf` along with `Iterator` in case of `ConcurrentModificationException`. Although I do not think that there is anything wrong with your code since you are calling them on 2 different lists, just check again to be safe so as not to introduce bugs.

Reply:
Hmm I see, I thought `removeIf` helped [prevent](https://www.baeldung.com/java-concurrentmodificationexception) `ConcurrentModificationException` so I used that instead, but using `remove` seems to work fine too so I'll change it.

Reply:
There is still a risk of `ConcurrentModificationException` as you are still modifying a `Collection` while iterating over it. `Iterators` actually help to prevent this. Saying this because I realised `removeIf` is invoked by a `Collection`.

Reply:
Mm I see, thanks!